src/server/object_services/map_builder.js

```diff
@@ -150,16 +156,33 @@ class MapBuilder {
             'chunks_to_delete_uniq', chunks_to_delete_uniq.length,
             'blocks_to_delete', blocks_to_delete.length);
 
+        // Delete parts only if this is eviction mode, and not move to tier
+        const delete_parts = (this.evict && !this.move_to_tier) ?
+                        map_deleter.delete_parts_by_chunks(chunks_to_delete_uniq) :
+                        Promise.resolve();
+
         await Promise.all([
+            delete_parts,
```

Comment:
usually it's easy enough to add such conditions inline and keep the "IO" flow simpler to read:

```suggestion
        await Promise.all([
            // Delete parts only if this is eviction mode, and not move to tier
            this.evict && map_deleter.delete_parts_by_chunks(chunks_to_delete_uniq),
```